pkg/thanos/statefulset.go

```diff
@@ -159,6 +160,29 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 	}
 	trEnvVars := []v1.EnvVar{}
 
+	// Use defaultReplicaLabelName constant by default if field is missing.
+	// Do not add label if field is set to empty string.
+	replicaLabelName := defaultReplicaLabelName
+	if tr.Spec.ReplicaLabelName != nil {
+		if *tr.Spec.ReplicaLabelName != "" {
+			replicaLabelName = *tr.Spec.ReplicaLabelName
+		} else {
+			replicaLabelName = ""
+		}
+	}
+	// TODO: Users can add other labels to ruler.
+	if replicaLabelName != "" {
+		trCLIArgs = append(trCLIArgs, fmt.Sprintf(`--label=%s="$(POD_NAME)"`, replicaLabelName))
+	}
+
+	// TODO: Users can add other alert drop labels to ruler.
+	dropLabels := []string{replicaLabelName}
+	for _, lb := range dropLabels {
+		if lb != "" {
+			trCLIArgs = append(trCLIArgs, fmt.Sprintf("--alert.label-drop=%s", lb))
```

Comment:
Is there any case that we don' t want to drop the replica label in the alerts? @pgier

Reply:
Not that I can think of.  Maybe @brancz or @simonpasquier have other ideas.

Reply:
I think we always want to drop them.